File: AlexaDeviceFinder-API/AlexaDeviceFinder-Auth/Models/AmazonUserDevice.cs
using System;
using Amazon.DynamoDBv2.DataModel;

namespace AlexaDeviceFinderAuth.Models
{
    [DynamoDBTable("AmazonUserDevices")]
    public class AmazonUserDevice
    {
        public enum DeviceOperatingSystem
        {
            Unknown,
            Android,
            Apple
        }

        [DynamoDBHashKey("AmazonUserID")]
        public string AmazonUserId { get; set; }

        [DynamoDBProperty("DeviceID")]
        public string DeviceId { get; set; }

        [DynamoDBProperty("DeviceOS")]
        public DeviceOperatingSystem DeviceOs { get; set; }

        [DynamoDBProperty("ModifiedDate")]
        public DateTime ModifiedDate { get; set; }

        public AmazonUserDevice()
        {
            this.ModifiedDate = DateTime.UtcNow;
        }

        public AmazonUserDevice(string userId, string deviceId)
        {
            this.AmazonUserId = userId;
            this.DeviceId = deviceId;
            this.ModifiedDate = DateTime.UtcNow;
        }

        public override string ToString()
        {
            return $"{this.AmazonUserId}:{ this.DeviceId}";
        }
    }
}

File: AlexaDeviceFinder-Android/app/src/main/java/com/jamespfluger/alexadevicefinder/auth/AmazonUserDevice.java
package com.jamespfluger.alexadevicefinder.auth;

public class AmazonUserDevice {
    private String userId;
    private String deviceId;
    private String deviceOs = "Android";

    public AmazonUserDevice(String userId, String deviceId) {
        this.userId = userId;
        this.deviceId = deviceId;
    }

    public String getUserId() {
        return this.userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getDeviceId() {
        return this.deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    public String getDeviceOs() { return this.deviceOs; }

    public void setDeviceOs(String deviceOs) { this.deviceOs = deviceOs; }

    @Override
    public String toString() {
        return userId + ":" + deviceId;
    }
}
